core/src/main/java/hudson/model/Computer.java

```diff
@@ -669,43 +670,47 @@ public boolean isLaunchSupported() {
     @Exported
     @Deprecated
     public boolean isTemporarilyOffline() {
-        return temporarilyOffline;
+        return getNodeOrDie().isTemporarilyOffline();
     }
 
     /**
      * @deprecated as of 1.320.
-     *      Use {@link #setTemporarilyOffline(boolean, OfflineCause)}
+     *      Use {@link #setTemporarilyOfflineCause(OfflineCause)}
      */
     @Deprecated
     public void setTemporarilyOffline(boolean temporarilyOffline) {
-        setTemporarilyOffline(temporarilyOffline, null);
+        setTemporarilyOfflineCause(temporarilyOffline ? new OfflineCause.LegacyOfflineCause() : null);
+    }
+
+    /**
+     * @deprecated as of TODO.
+     *      Use {@link #setTemporarilyOfflineCause(OfflineCause)} instead.
+     */
+    @Deprecated
+    public void setTemporarilyOffline(boolean temporarilyOffline, OfflineCause cause) {
+        offlineCause = temporarilyOffline ? cause : null;
     }
 
     /**
      * Marks the computer as temporarily offline. This retains the underlying
      * {@link Channel} connection, but prevent builds from executing.
      *
-     * @param cause
-     *      If the first argument is true, specify the reason why the node is being put
-     *      offline.
+     * @param temporarilyOfflineCause The reason why the node is being put offline.
+     *                                If null, this cancels the status
      */
-    public void setTemporarilyOffline(boolean temporarilyOffline, OfflineCause cause) {
-        offlineCause = temporarilyOffline ? cause : null;
-        this.temporarilyOffline = temporarilyOffline;
-        Node node = getNode();
-        if (node != null) {
-            node.setTemporaryOfflineCause(offlineCause);
-        }
-        synchronized (statusChangeLock) {
-            statusChangeLock.notifyAll();
-        }
-        if (temporarilyOffline) {
-            Listeners.notify(ComputerListener.class, false, l -> l.onTemporarilyOffline(this, cause));
+    public void setTemporarilyOfflineCause(OfflineCause temporarilyOfflineCause) {
+        getNodeOrDie().setTemporaryOfflineCause(temporarilyOfflineCause);
+        if (temporarilyOfflineCause != null) {
+            Listeners.notify(ComputerListener.class, false, l -> l.onTemporarilyOffline(this, temporarilyOfflineCause));
         } else {
             Listeners.notify(ComputerListener.class, false, l -> l.onTemporarilyOnline(this));
         }
```

Comment:
The field `offlineCause` is only for technical offline cause now. `temporarilyOfflineCause` only exists in `Node` now.